cmd/finch/nerdctl.go

```diff
@@ -142,95 +142,224 @@ func (nc *nerdctlCommand) run(cmdName string, args []string) error {
 		}
 	}
 
-	// envArgPos is used to preserve the position of first environment parameter
-	envArgPos = -1
-	// if a debug flag is passed before env arg pos we reduce the env arg pos by 1 to account for skipping debug flag
-	isDebug = 0
-	for i, arg := range args {
-		// Check if command requires arg handling
-		if hasArgHandler {
-			// Check if argument for the command needs handling, sometimes it can be --file=<filename>
-			b, _, _ := strings.Cut(arg, "=")
-			h, ok := aMap[b]
-			if ok {
-				err = h(nc.systemDeps, args, i)
-				if err != nil {
-					return err
-				}
-				// This is required when the positional argument at i is mutated by argHandler, eg -v=C:\Users:/tmp:ro
-				arg = args[i]
-			}
-		}
-		// parsing environment values from the command line may pre-fetch and
-		// consume the next argument; this loop variable will skip these pre-consumed
-		// entries from the command line
-		if skip {
-			skip = false
-			continue
-		}
+	switch cmdName {
+	case "container", "run", "exec", "compose", "container run":
+		// check if an option flag is present; immediately following the command
 		switch {
-		case arg == "--debug":
-			// explicitly setting log level to avoid `--debug` flag being interpreted as nerdctl command
-			if envArgPos == -1 {
-				isDebug = 1
-			}
-			nc.logger.SetLevel(flog.Debug)
-		case argIsEnv(arg):
-			if envArgPos == -1 {
-				envArgPos = i - isDebug
-			}
-			shouldSkip, addEnv := handleEnv(nc.systemDeps, arg, args[i+1])
-			skip = shouldSkip
-			if addEnv != "" {
-				envs = append(envs, addEnv)
+		case args[0] == "run" && strings.HasPrefix(args[1], "-"):
+			firstOptPos = 1
+		case strings.HasPrefix(args[0], "-"):
+			firstOptPos = 0
+		default:
+			firstOptPos = -1
+		}
+		// lastOpt is used to track when no more flags need to be processed
+		lastOpt = false
+
+		shortFlagBoolSet := sets.NewString("-d", "-i", "-t")
+		longFlagBoolSet := sets.NewString(
+			"--detach", "--init", "--interactive", "--oom-kill-disable",
+			"--privileged", "--read-only", "--rm", "--rootfs", "--tty")
+
+		shortFlagArgsSet := sets.NewString("-e", "-h", "-m", "-u", "-w", "-p", "-l", "-v")
+
+		for i, arg := range args {
+			// Check if command requires arg handling
+			if hasArgHandler {
```

Comment:
Thanks for the comment. I believe that you are highlighting the precedence of the cmdHandler and argHandler paradigm.

After finding the nerdctl_windows implementation, I explored the current mechanisms for inspiration for my current change set. My primary observation was that the model was primarily positioned for "in place" argument manipulation/updating. Whereas, the actual argument&value processing (passing to resulting runArguments) was wholly contained here.

I agree that there is an optimization case for "command specific argument&value processing", specifically the possibility of cmdFlagSets and generalized processing helper functions.
In this case, I have attempted to deliver a "small unit" incremental change to address the scope of the Issue description, rather than a drastic refactor.

My next iteration leverages a bit of generalization for the FlagSets and an improved "short flag group" handler that will set the stage for another incremental refactor for further command specific argument&value processing.

Reply:
Thanks, the code is a lot more readable now